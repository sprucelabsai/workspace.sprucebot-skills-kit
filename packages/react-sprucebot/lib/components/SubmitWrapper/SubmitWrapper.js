"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var SubmitWrapper = function SubmitWrapper(_ref) {
  var className = _ref.className,
      props = (0, _objectWithoutProperties2.default)(_ref, ["className"]);
  return _react.default.createElement("div", (0, _extends2.default)({}, props, {
    className: "submit__wrapper ".concat(className || '')
  }));
};

var _default = SubmitWrapper;
exports.default = _default;