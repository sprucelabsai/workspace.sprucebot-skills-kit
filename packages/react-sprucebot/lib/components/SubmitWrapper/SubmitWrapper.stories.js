'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _SubmitWrapper = require('./SubmitWrapper');

var _SubmitWrapper2 = _interopRequireDefault(_SubmitWrapper);

var _SubmitWrapper3 = require('./SubmitWrapper.md');

var _SubmitWrapper4 = _interopRequireDefault(_SubmitWrapper3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('SubmitWrapper', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_SubmitWrapper4.default, function () {
  return _react2.default.createElement(_SubmitWrapper2.default, null);
}));