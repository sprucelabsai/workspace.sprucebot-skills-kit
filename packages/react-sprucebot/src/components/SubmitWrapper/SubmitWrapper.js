import React from 'react'

const SubmitWrapper = ({ className, ...props }) => {
	return <div {...props} className={`submit__wrapper ${className || ''}`} />
}

export default SubmitWrapper
