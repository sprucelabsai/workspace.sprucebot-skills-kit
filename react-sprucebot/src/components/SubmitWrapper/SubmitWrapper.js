import styled from 'styled-components'

const SubmitWrapper = styled.div.attrs({
	className: 'submit__wrapper'
})``

export default SubmitWrapper
