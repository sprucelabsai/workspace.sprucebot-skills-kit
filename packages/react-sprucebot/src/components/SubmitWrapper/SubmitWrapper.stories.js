import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import SubmitWrapper from './SubmitWrapper'
import Button from '../Button/Button'
import readme from './SubmitWrapper.md'

const stories = storiesOf('SubmitWrapper', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, () => (
		<SubmitWrapper>
			<Button primary>Button 1</Button>
			<Button alt>Button 2</Button>
			<Button alt>Button 3</Button>
		</SubmitWrapper>
	))
)
