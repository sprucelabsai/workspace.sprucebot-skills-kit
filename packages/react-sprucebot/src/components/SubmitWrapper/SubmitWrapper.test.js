import React from 'react'
import SubmitWrapper from './SubmitWrapper'
import { shallow } from 'enzyme'

describe('SubmitWrapper tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<SubmitWrapper />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
