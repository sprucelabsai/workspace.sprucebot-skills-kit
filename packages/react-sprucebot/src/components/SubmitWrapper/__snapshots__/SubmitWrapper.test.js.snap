// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="SubmitWrapper-s5rqw4n-0 submit__wrapper jWcvtf"
/>
`;
